0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md

```diff
@@ -167,14 +196,40 @@ Content-Type: application/spsp4+json
 }
 ```
 
-### Payment Setup
+## Model of Operation
+
+We assume that the client knows the server's SPSP endpoint (see [Payment Pointers](../0026-payment-pointers/0026-payment-pointers.md)).
+
+1. The user's SPSP client queries the server's SPSP Endpoint.
+
+2. The SPSP endpoint responds with the server info, including the servers's ILP address (`destination_account`) and the shared secret (`shared_secret`) to be used in STREAM. 
+    * The `destination_account` SHOULD be used as the STREAM destinationAccount.
+    * The `shared_secret` SHOULD be decoded from base64 and used as the STREAM sharedSecret.
+  
+    It MAY respond with additional information if it is an invoice server or a pull payment server. If the amounts within `balance` are negative, the client can push to this endpoint. If the amounts within `balance` are positive, the client can pull from this endpoint. If `balance` is not provided, only push payments are possible. 
+
+3. The SPSP client establishes a STREAM connection to the server using the server's ILP address and shared secret.
+
+In case of push payments or invoices, the process continues as follows:
 
-The sender uses the receiver details to create the STREAM connection:
+4. The SPSP client begins sending ILP packets to fulfill the payment.
+    1. The client will construct a stream to the server on the ILP/STREAM connection.
+    2. The client will adjust their `sendMax` to reflect the amount they're willing to send.
+        * If present, `|balance.current|` SHOULD be used as the STREAM `sendMax`.
+    3. The server will adjust their `receiveMax` to reflect the amount they're willing to receive.
+        * If present, `|balance.current|` SHOULD be used as the STREAM `receiveMax`.
+    4. The client's and server's STREAM modules will move as much value as possible while staying inside these bounds.
+    5. If the client reaches their `sendMax`, they end the stream and the connection. If the server reaches their `receiveMax`, they will end the stream and the connection.
 
-* The `destination_account` should be used as the STREAM destinationAccount.
-* The `shared_secret` should be decoded from base64 and used as the STREAM sharedSecret.
-* If present, the `balance.maximum - balance.current` SHOULD be used as the STREAM `receiveMax`.
+In case of pull payments, the process continues as follows:
 
-In a UI, the `asset_info` and `receiver_info` objects (if present) can be used for display purposes. These objects can be manipulated by the receiver in any way, so amounts SHOULD be displayed in source units when possible.
+4. The SPSP server begins sending ILP packets to fulfill the payment.
```

Comment:
Would it instead be the client constructing the stream to initiate the pull payment?